pkg/registry/registry.go

```diff
@@ -0,0 +1,32 @@
+// Package registry is used to keep all the module, runtime, backend and the corresponding
+// config structure, so that the workspace config can be correctly deserialized.
+package registry
+
+import (
+	"kusionstack.io/kusion/pkg/models"
+	"kusionstack.io/kusion/pkg/models/appconfiguration/accessories/database"
+)
+
+var (
+	registeredModules  []models.Module
```

Comment:
The module code might be put out side of Kusion repo, so how to register modules here?

Reply:
The registry pattern is based on the assumption that Kusion can get all the module structure. For the module code  put outside kusion repo, the thing goes the same, kusion has to get the module structure, whatever the way.
Of course, there is more work, and the exact work is unknown. It depends on the new generator scheme.
Hence, it seems that forgetting about the registry, and make it simple is more suitable for now.